src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -39,25 +39,25 @@ public ManagedCompiler()
         #region Properties
 
         // Please keep these alphabetized.
-        public string[] AdditionalLibPaths
+        public string[]? AdditionalLibPaths
         {
             set { _store[nameof(AdditionalLibPaths)] = value; }
             get { return (string[])_store[nameof(AdditionalLibPaths)]; }
```

Comment:
:memo: This appears to a bug/limitation somewhere. Casting to array types is not catching the issue the same way casting to `string` below was.

Reply:
For some reason, I remember this is by-design. The cast to `(string)` doesn't change anything. On the other hand, the cast to `(string?)` affects the declared nullability of the expression (so `var x = (string?)y;` gives `var` a `string?` type).

@cston to confirm.

Reply:
Why does it allow casting a nullable object to a non-nullable type without `!`?

Reply:
A warning should be reported. With the current `features/NullableReferenceTypes`, the following generates `warning CS8600: Converting null literal or possible null value to non-nullable type`:
```
    Dictionary<string, object?> _store = new Dictionary<string, object?>();
    string[]? AdditionalLibPaths
    {
        get { return (string[])_store[nameof(AdditionalLibPaths)]; }
    }
```